File: MoviesMVVMAPPWithPagingJava/app/src/main/java/com/masterclass/moviesmvvmappwithpagingjava/api/APIInterface.java
package com.masterclass.moviesmvvmappwithpagingjava.api;

import com.masterclass.moviesmvvmappwithpagingjava.models.MoviesResponse;

import io.reactivex.rxjava3.core.Single;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface APIInterface {
    @GET("movie/popular")
    Single<MoviesResponse> getMoviesByPage(@Query("page") int page);
}

File: MoviesMVVMAPPWithPagingKotlin/app/src/main/java/com/masterclass/moviesmvvmappwithpagingkotlin/api/APIInterface.kt
package com.masterclass.moviesmvvmappwithpagingkotlin.api

import com.masterclass.moviesmvvmappwithpagingkotlin.models.MoviesResponse
import io.reactivex.rxjava3.core.Single
import retrofit2.http.GET
import retrofit2.http.Query

interface APIInterface {
    @GET("movie/popular")
    fun getMoviesByPage(
        @Query("page") page: Int
    ): Single<MoviesResponse>
}